File: easy/3940_limit_occurrences_in_sorted_array/solution.cpp
#include <vector>

class Solution {
public:
  std::vector<int> limitOccurrences(std::vector<int> &nums, int k) {
    std::vector<int> limitedNums = {nums[0]};

    int count = 1;

    for (size_t i = 1; i < nums.size(); ++i) {
      if (limitedNums.back() == nums[i]) {
        if (count == k)
          continue;

        ++count;
      } else
        count = 1;

      limitedNums.push_back(nums[i]);
    }

    return limitedNums;
  }
};

File: easy/3940_limit_occurrences_in_sorted_array/solution.py
from typing import List


class Solution:
    def limitOccurrences(self, nums: List[int], k: int) -> List[int]:
        limitedNums = [nums[0]]

        i = 1
        count = 1
        curr = nums[0]

        while i < len(nums):
            if nums[i] != curr:
                curr = nums[i]
                count = 1
            elif count == k:
                i += 1
                continue
            else:
                count += 1

            limitedNums.append(curr)
            i += 1

        return limitedNums
